File: binance4j-websocket/src/main/java/com/binance4j/websocket/serialization/AssetBalanceDeserializer.java
package com.binance4j.websocket.serialization;

import java.io.IOException;

import com.binance4j.core.dto.AssetBalance;
import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.JsonDeserializer;
import com.fasterxml.jackson.databind.JsonNode;

/** {@link AssetBalance} deserializer */
public class AssetBalanceDeserializer extends JsonDeserializer<AssetBalance> {
	@Override
	public AssetBalance deserialize(JsonParser jp, DeserializationContext ctx) throws IOException {
		JsonNode n = jp.getCodec().readTree(jp);
		return new AssetBalance(n.get("a").asText(), n.get("f").asText(), n.get("l").asText());
	}
}

File: connectors/src/main/kotlin/com/binance4j/websocket/serialization/AssetBalanceDeserializer.kt
package com.binance4j.websocket.serialization

import com.binance4j.core.dto.AssetBalance
import com.fasterxml.jackson.core.JsonParser
import com.fasterxml.jackson.databind.DeserializationContext
import com.fasterxml.jackson.databind.JsonDeserializer
import com.fasterxml.jackson.databind.JsonNode
import java.io.IOException

/** [AssetBalance] deserializer  */
class AssetBalanceDeserializer : JsonDeserializer<AssetBalance>() {
    @Throws(IOException::class)
    override fun deserialize(jp: JsonParser, ctx: DeserializationContext): AssetBalance {
        val n = jp.codec.readTree<JsonNode>(jp)
        return AssetBalance(n["a"].asText(), n["f"].asText(), n["l"].asText())
    }
}